src/edgepi/reg_helper/reg_helper.py

```diff
@@ -116,11 +115,13 @@ def apply_opcodes(register_values: dict, opcodes: list):
             register_entry["value"] = _apply_opcode(register_entry["value"], opcode)
             register_entry["is_changed"] = True
 
-    __validate_register_updates(original_regs, register_values)
+    # NOTE: disabling register update validation because there's no reason for us to suspect the other 
+    # values would change? I'm not sure why this was done before.
+    # This is also quite slow, and affects the performance significantly
+    #__validate_register_updates(original_regs, register_values)
```

Comment:
this validation function, combined with the deepcopy above is quite slow, and I can't figure out if it actually does anything. If it does, I'll probably want to make a function `unsafe_apply_opcodes()` which removes it, so that the rest of the codebase isn't adversely affected.